lrlex/src/lib/lexer.rs

```diff
@@ -20,12 +20,15 @@ use crate::{
 
 #[doc(hidden)]
 /// Corresponds to the options for `regex::RegexBuilder`.
-#[derive(Clone)]
-pub struct RegexOptions {
-    pub dot_matches_new_line: bool,
-    pub multi_line: bool,
-    pub octal: bool,
-    pub posix_escapes: bool,
+#[derive(Clone, Debug)]
+pub struct LexFlags {
+    // The following values when `None` grmtools provides default values for in `DEFAULT_LEX_FLAGS`
+    pub dot_matches_new_line: Option<bool>,
+    pub multi_line: Option<bool>,
```

Comment:
One thing we could consider here is using the `Default` trait,

We kind of have the same thing with the `const UNSPECIFIED_LEX_FLAGS` though.
the way construct this in a way that allows us to add fields in the future is similar to how the Default trait is often used.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=028e54b786184a9d7f4db48a82aabd88

Reply:
Anyhow, I think we should try and make this  `#[non_exhaustive] struct LexFlags`,
It seems that precludes functional record update, but I think currently works/forces
the pattern i've been using of.

```
let mut x = UNSPECIFIED_LEX_FLAGS;
x.flag = Some(...)
````

Reply:
Should be fixed in [d5bfe8a](https://github.com/softdevteam/grmtools/pull/496/commits/d5bfe8a411d1591577d482f579f78352c5ba904a)

Reply:
`non_exhaustive` is a new one on me, but having look it up, I agree that it seems like a good idea here. [I wonder if there are other places we should use this in the public(ish) API?]

Reply:
I think `LexErrorKind` and `YaccGrammarErrorKind` are both probably good candidates for `non_exhaustive`.